test/serialize-response.test.js

```diff
@@ -106,119 +106,79 @@ const handler = (request, reply) => {
   })
 }
 
-test('serialize the response for a Bad Request error, as defined on the schema', t => {
+test('serialize the response for a Bad Request error, as defined on the schema', async t => {
   const fastify = Fastify({})
 
-  t.teardown(fastify.close.bind(fastify))
-
   fastify.post('/', options, handler)
+  const response = await fastify.inject({
+    method: 'POST',
+    url: '/'
+  })
 
-  fastify.listen({ port: 0 }, err => {
-    t.error(err)
-
-    const url = `http://localhost:${fastify.server.address().port}/`
-
-    sget({
-      method: 'POST',
-      url,
-      json: true
-    }, (err, response, body) => {
-      t.error(err)
-      t.equal(response.statusCode, 400)
-      t.same(body, {
-        statusCode: 400,
-        error: 'Bad Request',
-        message: 'body must be object'
-      })
-      t.end()
-    })
+  t.equal(response.statusCode, 400)
+  t.same(sjson(response.body), {
+    statusCode: 400,
+    error: 'Bad Request',
+    message: 'body must be object'
   })
+  t.end()
```

Comment:
Can you specify `t.plan(<number>)` instead of using `t.end()`

Reply:
DONE